clang/test/SemaSYCL/kernelname-enum.cpp

```diff
@@ -1,13 +1,42 @@
-// RUN: %clang_cc1 -I %S/Inputs -fsycl -fsycl-is-device -fsycl-int-header=%t.h -fsyntax-only -Wno-sycl-2017-compat -verify %s
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsycl-int-header=%t.h -fsyntax-only -Wno-sycl-2017-compat -verify %s
 
-#include "sycl.hpp"
+//#include "Inputs/sycl.hpp"
+#ifndef SYCL_HPP
+#define SYCL_HPP
+
+class auto_name {};
+template <typename Name, typename Type>
+struct get_kernel_name_t {
+  using name = Name;
+};
+template <typename Type>
+struct get_kernel_name_t<auto_name, Type> {
+  using name = Type;
+};
+#define ATTR_SYCL_KERNEL __attribute__((sycl_kernel))
+template <typename KernelName = auto_name, typename KernelType>
+ATTR_SYCL_KERNEL void kernel_single_task(const KernelType &kernelFunc) {
+  kernelFunc();
+}
+  template <typename KernelName = auto_name, typename KernelType>
+  void single_task(const KernelType &kernelFunc) {
+    using NameT = typename get_kernel_name_t<KernelName, KernelType>::name;
+#ifdef __SYCL_DEVICE_ONLY__
+// expected-error@+1 1+{{kernel name is invalid. Unscoped enum requires fixed underlying type}}
+    kernel_single_task<NameT>(kernelFunc);
+#else
+    kernelFunc();
+#endif
+  }
+
+#endif
 
 enum unscoped_enum_int : int {
   val_1,
   val_2
 };
 
-// expected-note@+1 {{'unscoped_enum_no_type_set' declared here}}
+// expected-note@+1 1+ {{'unscoped_enum_no_type_set' declared here}}
```

Comment:
What is "1+"? I've never seen this before.

Reply:
It is used to indicate the number of times this error is expected. In our case it is 2, since I invoke the kernel-caller-function twice for f2 and f5

Reply:
From : https://clang.llvm.org/doxygen/classclang_1_1VerifyDiagnosticConsumer.html
Where the diagnostic is expected to occur a minimum number of times, this can be specified by appending a '+' to the number. Example:

void f(); // expected-note 0+ {{previous declaration is here}}
void g(); // expected-note 1+ {{previous declaration is here}}

Reply:
Could you please specify it explicitly? i.e. just '2'